test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -507,6 +507,46 @@ TEST_F(ClientLbEnd2endTest, PickFirstResetConnectionBackoff) {
   EXPECT_LT(waited_ms, kInitialBackOffMs);
 }
 
+TEST_F(ClientLbEnd2endTest,
+       PickFirstResetConnectionBackoffNextAttemptStartsImmediately) {
+  ChannelArguments args;
+  constexpr int kInitialBackOffMs = 1000;
+  args.SetInt(GRPC_ARG_INITIAL_RECONNECT_BACKOFF_MS, kInitialBackOffMs);
+  const std::vector<int> ports = {grpc_pick_unused_port_or_die()};
+  auto channel = BuildChannel("pick_first", args);
+  auto stub = BuildStub(channel);
+  SetNextResolution(ports);
+  // Wait for connect, which should fail ~immediately, because the server
+  // is not up.
+  gpr_log(GPR_INFO, "=== INITIAL CONNECTION ATTEMPT");
+  EXPECT_FALSE(
+      channel->WaitForConnected(grpc_timeout_milliseconds_to_deadline(10)));
+  // Reset connection backoff.
+  gpr_log(GPR_INFO, "=== RESETTING BACKOFF");
+  experimental::ChannelResetConnectionBackoff(channel.get());
+  // Trigger a second connection attempt.  This should also fail
+  // ~immediately, but the retry should be scheduled for
+  // kInitialBackOffMs instead of applying the multiplier.
+  gpr_log(GPR_INFO, "=== TRIGGERING SECOND CONNECTION ATTEMPT");
+  EXPECT_FALSE(
+      channel->WaitForConnected(grpc_timeout_milliseconds_to_deadline(10)));
+  // Bring up a server on the chosen port.
+  gpr_log(GPR_INFO, "=== STARTING BACKEND");
+  StartServers(1, ports);
+  // Wait for connect.  Should happen within kInitialBackOffMs.
+  gpr_log(GPR_INFO, "=== TRIGGERING THIRD CONNECTION ATTEMPT");
+  const gpr_timespec t0 = gpr_now(GPR_CLOCK_MONOTONIC);
```

Comment:
To make it safer, I think `t0` can be recorded right before line 526, since that seems the minimal closure to cover the initial backoff interval. Otherwise, if line 537 is taking 1s, we can't prove the fix.

Reply:
The time that the timer is scheduled is not that important. It's the time that the next attempt time is determined that is important. Hmm.. I know this sounds confusing. 

Actually, when we start the second connection attempt in step 2, we determine the next attempt time (a deadline), which is now + initial backoff. https://github.com/grpc/grpc/blob/e97c9457e2f4e6733873ea2975d3b90432fdfdc1/src/core/ext/filters/client_channel/subchannel.cc#L667 

So I think we should record `t0` right before step 2.

Reply:
Good catch!  You're right, I'd forgotten that we computed the start time of the next attempt before starting the previous one.

Done.